File: src/main/java/szewek/flux/recipe/CompactingRecipe.java
package szewek.flux.recipe;

import net.minecraft.item.ItemStack;
import net.minecraft.item.crafting.IRecipeSerializer;
import net.minecraft.util.ResourceLocation;
import szewek.flux.F;

public final class CompactingRecipe extends AbstractMachineRecipe {
	public CompactingRecipe(ResourceLocation idIn, String groupIn, MachineRecipeSerializer.Builder builder) {
		super(F.R.COMPACTING, idIn, groupIn, builder);
	}

	public ItemStack getIcon() {
		return new ItemStack(F.B.COMPACTOR);
	}
}

File: src/main/java/szewek/flux/recipe/CompactingRecipe.kt
package szewek.flux.recipe

import net.minecraft.item.ItemStack
import net.minecraft.item.crafting.IRecipeSerializer
import net.minecraft.util.ResourceLocation
import szewek.flux.FBlocks
import szewek.flux.FRecipes

class CompactingRecipe(idIn: ResourceLocation, groupIn: String, builder: MachineRecipeSerializer.Builder)
    : AbstractMachineRecipe(FRecipes.COMPACTING, idIn, groupIn, builder) {
    override fun getIcon(): ItemStack {
        return ItemStack(FBlocks.COMPACTOR)
    }

    override fun getSerializer(): IRecipeSerializer<*> = FRecipes.COMPACTING_SERIALIZER
}